Ensure we get the interpreter from python-any-r1 rather than same
as meson. Plus find_program() is quicker than find_installation(),
and the latter also needs distutils that is unavailable with
python3_12 without setuptools.

https://bugs.gentoo.org/731728
--- a/meson.build
+++ b/meson.build
@@ -323,3 +323,3 @@
 thirdparty = meson.project_source_root()/'3rdparty'
-python = import('python').find_installation()
+python = find_program('python3')
 python_env = environment()
